app/jobs/update_late_tasks_job.rb

```diff
@@ -0,0 +1,8 @@
+class UpdateLateTasksJob < ApplicationJob
+  queue_as :default
+
+  def perform(*args)
+    late_tasks = Task.where("due_date < ?", Date.today)
```

Comment:
Get all tasks that match the due date criteria in a single query, rather than looping through all tasks and performing the check in memory.